fetch/mesh_data.go

```diff
@@ -97,21 +102,24 @@ func (f *Fetch) getHashes(
 				options.limiter.Release(1)
 				pendingMetric.Add(-1)
 				if p.err != nil {
-					f.logger.Debug("failed to get hash",
+					f.logger.With().Debug("failed to get hash",
 						log.String("hint", string(hint)),
 						log.Stringer("hash", h),
 						log.Err(p.err),
 					)
-					failed.Add(1)
+
+					mu.Lock()
+					bfailure.Add(h, p.err)
```

Comment:
You can use `errors.Join` instead:

```go
	var (
		eg       errgroup.Group
		mu       sync.Mutex
		errBatch error
	)
	// ...
	for i, hash := range hashes {
		// ...
		if p.err != nil {
			f.logger.Debug("failed to get hash",
			f.logger.With().Debug("failed to get hash",
				log.String("hint", string(hint)),
				log.Stringer("hash", h),
				log.Err(p.err),
			)
			
			mu.Lock()
			errors.Join(errBatch, p.err)
			mu.Unlock()
	// ...
	eg.Wait()
	return errBatch // is nil if no errors happened
}
```

Reply:
thanks, i wasn't aware of this. but i also need to check which hash failed on the caller side, hence the map in the BatchError.

Reply:
i opened an issue to refactor it, will do today/monday

i added this error in the original change that i was merging yesterday, when we talked about it yesterday